### ISSUE TITLE
Metasploit path?

### ISSUE THREAD

**Issue by EchoUnderscoreNine** [trustedsec/social-engineer-toolkit#315]
I tried setting the METASPLOIT_PATH  ---> =/opt/metasploit-framework in the set.config file also other varients but nothing seems to work & keep getting metasploit not found ! any ideas ?

BTW i`m using ubuntu 16.04.

thx in adv.

**Comment by Excelsor** [comment]
I am using Macos Sierra, and... the Metasploit v. 4.12 is running very well, but the Setoolkit don't reconize the Metasploit PATH.

I tryed change the set.config unsuccessful on the /etc/setoolkit paste pressing:
`sudo nano set.config`

I installed: 
Setoolkit v 7.4.1 (working without the exploits)
Metasploit v 4.12 (working)
Python3 (working)

Help?

**Comment by EchoUnderscoreNine** [comment]
thx for your time to answer & my late reply, but still not working. I haven`t installed msf from rapid7 it`s 

metasploit v4.12.35-dev-74340e9 - bleeding edge + i don`t have a /bin folder

any remarks or help would be highly appreciated.
thx for yr time.

**Comment by Excelsor** [comment]
Put in the set.config the folder where the Metasploit (msfconsole, msfvenom) are installed. I don't know where is in the Ubuntu. That version 4 of Metasploit in the Linux probably it's on the /opt/metasploit-framework.

So try:
Define the path to Metasploit. For example, /opt/metasploit-framework [PRESS ENTER HERE]
METASPLOIT_PATH=/opt/metasploit-framework

After: Control + X , press 'y' to save and try run the Setoolkit.

I hope help! @Excelsor

**Comment by EchoUnderscoreNine** [comment]
I presume you mean to edit the set.config via nano .. tried that already .. strange, it worked on the previous release though this one seems all intact, it`s just not working!

thx anyway .. cheers.

**Comment by trustedsec** [comment]
You need to point the path to wherever msfconsole is. When installing Metasploit. Can just find / -name msfconsole. If its under /opt/metasploit-framework - then I wouldn't know what the issue is here - when I move directory paths and give it a different option in set.config - it works without an issue. 

Normally Metasploit is installed under:

/opt/metasploit-framework/bin/msfconsole

As an example - I moved my /opt/metasploit-framework directory to /opt/test

Edited the set.config -> put the metasploit path to /opt/test/bin

No issues - launched and detected fine and returned right path.

**Comment by EchoUnderscoreNine** [comment]
well its kinda odd but it looks like ubuntu is trying to sanitize it self, making installing other sources impossible so i moved to debian & did a normal clone, installed and bobs your uncle everything works just fine !! once again thanks to all you guys & thank you T.Sec for a great piece of art.

**Comment by trustedsec** [comment]
Thanks! =) Appreciate it. I'll take a peek on the Ubuntu front. I'll consider this closed. Appreciate the report and glad it worked out!

**Comment by H011YxW00D** [comment]
So in setcore.py starting on line 313 there are hard coded entries that is SET is searching for for MSF. Since I am using Kubuntu Linux as well as the binary installed version from rapid 7 I edited the backbox entry to look like this:

 if os.path.isfile("/opt/metasploit-framework/app/msfconsole"):
            msf_path = "/opt/metasploit-framework/"
            trigger = 1

Then I edited my set.config to point to /opt/metasploit-framework and it fired up without issue. Like others have said it seems to just want to know where the msfconsole binary is. I also experimented with copying the msfconsole link under /opt/metasploit/app to /opt/metasploit-framework/ and that worked as well.
